### PR TITLE
Added log4net support.

### PR DESCRIPTION
**Pull request by teamagam3**

### CODE REVIEW

**Comment by review-ninja** [comment]
<a href="https://app.review.ninja/teamagam/GGSmtp/pull/5" target="_blank">ReviewNinja

File: GGSmtp/GGSmtp.ServerApp/MailServerLog4Net.cs [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+using Simple.MailServer.Logging;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace GGSmtp.ServerApp
+{
+    public class MailServerLog4Net : IMailServerLogger
```
Comment (BarAgam) [inline GGSmtp/GGSmtp.ServerApp/MailServerLog4Net.cs]: Log4NetMailServerLogger ? :)

File: GGSmtp/GGSmtp.ServerApp/Program.cs [hunk 2]
```diff
@@ -14,8 +14,9 @@ public static void Main(string[] args)
         {
             short port = Settings.Default.HostPort;
             MailServerLogLevel verbosity = Settings.Default.LogVerbosity;
+            MailServerLog4Net logger = new MailServerLog4Net(verbosity);
```
Comment (BarAgam) [inline GGSmtp/GGSmtp.ServerApp/Program.cs]: See, you call it a logger here

File: GGSmtp/GGSmtp.ServerApp/MailServerLog4Net.cs [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+        {
+            get
+            {
+                return mLogLevel;
+            }
+        }
+
+        public void Debug(string message)
+        {
+            if (mLogLevel >= MailServerLogLevel.Debug)
```
Comment (BarAgam) [inline GGSmtp/GGSmtp.ServerApp/MailServerLog4Net.cs]: refactor the if condition out to avoid code duplication throughout the class !fix

Comment (BarAgam) [inline GGSmtp/GGSmtp.ServerApp/MailServerLog4Net.cs]: boolean shouldLog(MailServerLogLevel level) ?

Comment (teamagam3) [inline GGSmtp/GGSmtp.ServerApp/MailServerLog4Net.cs]: !fixed

**Pull request merged**